python/tvm/relay/backend/contrib/ethosu/tir/scheduler.py

```diff
@@ -135,7 +135,17 @@ def _visit(tensor, reader, lut):
             if tensor not in planned:
                 planned.add(tensor)
                 if isinstance(tensor.op, tvm.te.PlaceholderOp) and tensor != lut:
-                    index = list(cached_func.inputs).index(tensor)
+                    # Find index of input using 'same_as' check to prevent equality
+                    # ambiguity when encountering a scalar.
+                    index = -1
+                    for i, var in enumerate(cached_func.inputs):
```

Comment:
nit : could use something like : 
[input for input in cached_func.inputs if input.same_as(tensor)] and then checking the resulting list ?

Reply:
Agreed, I'll make the change here :)